config/config.go

```diff
@@ -120,23 +225,19 @@ func init() {
 // It's supposed to be called after init().
 // The configDirPath argument can be given to override the DefaultConfigDirPath.
 func Initialize(configDirPath string) error {
+	// Reset our Config object. This is particularly necessary for running
+	// multiple tests in a row.
+	Config = configStruct{}
```

Comment:
I really dislike this is a global variable. This also makes `WriteToFile` very magical.

Reply:
What would be a better way?

Reply:
I checked out the code.
What is confusing that there are methods of the config struct itself, functions (i.e. `WriteToFile`) which rely on the fact that `config.Config` is a global variable.

I'd be ok for me if either `WriteToFile` took a config argument or config struct had method `WriteToFile`.